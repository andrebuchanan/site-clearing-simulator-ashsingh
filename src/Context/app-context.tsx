import React, { ReactNode, useMemo, useState } from 'react'

interface AppContextInterface {
  loggedCommands: string[]
  logCommand: (command: string) => void
}

export const AppContext = React.createContext<AppContextInterface>({
  loggedCommands: [],
  logCommand: () => undefined
})

export const AppContextProvider = ({ children }: { children: ReactNode }) => {
  const [commands, setCommands] = useState<string[]>([])

  const logCommandHandler = (command: string) => {
    setCommands((prevState) => [...prevState, command])
  }

  const contextValue = useMemo(
    () => ({ loggedCommands: commands, logCommand: logCommandHandler }),
    [commands]
  )

  return (
    <AppContext.Provider value={contextValue}>{children}</AppContext.Provider>
  )
}

export default AppContextProvider
