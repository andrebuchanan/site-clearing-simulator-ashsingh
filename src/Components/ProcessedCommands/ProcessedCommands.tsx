import React, { useContext } from 'react'
import uniqid from 'uniqid'

import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Typography from '@mui/material/Typography'

import { AppContext } from '../../Context'

export const ProcessedCommands = () => {
  const { loggedCommands } = useContext(AppContext)
  return (
    <>
      <Typography variant="h6" component="div" gutterBottom>
        Processed Commands
      </Typography>
      <List>
        {loggedCommands.map((command) => (
          <ListItem key={uniqid()} disablePadding>
            <ListItemText primary={command} />
          </ListItem>
        ))}
      </List>
    </>
  )
}

export default ProcessedCommands
