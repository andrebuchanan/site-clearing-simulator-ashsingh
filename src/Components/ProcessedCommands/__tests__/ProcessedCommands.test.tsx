import React from 'react'
import { render, screen } from '@testing-library/react'

import { AppContext } from '../../../Context'
import ProcessedCommands from '../ProcessedCommands'

const contextValue = {
  loggedCommands: ['ash', 'wini'],
  logCommand: () => undefined
}

test('renders App', () => {
  const { asFragment } = render(
    <AppContext.Provider value={contextValue}>
      <ProcessedCommands />
    </AppContext.Provider>
  )

  expect(asFragment()).toMatchSnapshot()
  expect(screen.getByText('ash')).toBeDefined()
})
