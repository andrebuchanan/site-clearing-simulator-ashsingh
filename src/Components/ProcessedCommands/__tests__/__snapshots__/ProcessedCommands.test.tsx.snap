// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
  >
    Processed Commands
  </div>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          ash
        </span>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          wini
        </span>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
